### ISSUE TITLE
Worker relative path resolution always relative to cwd

### ISSUE THREAD

**Issue by kevinkassimo** [denoland/deno#2745]
This means for nested workers the path would still be relative to cwd, causing surprising behavior.

Current behavior example, running `deno -A ./main.js`:

`./main.js`:
```js
new Worker("./workers/a.js");
```

`./workers/a.js`:
```js
import "./mod.js"; // NOTICE that this is still working as expected.

console.log("a.js");
new Worker("./workers/b.js"); // NOTICE that this has to relative to CWD, aka `.` in our context, instead of `./b.js`
// This actually impacts all relative paths except for ones used in import statements.
```

`./workers/b.js`:
```js
console.log("b.js");
```

`./workers/mod.js`:
```js
console.log("mod.js");
```

Current behavior is at odds with static import module resolution as shown above.

Prior discussion in Node: https://github.com/nodejs/node/issues/21686

This behavior does not seem to happen in Chrome, in which case `new Worker("./b.js");` would work correctly. (also Chrome still hides `type: "module"` behind a flag: https://crbug.com/680046)

(I personally believe Chrome's behavior is less surprising and allows the illusion that a `Worker` is its own entity, and I think making people confident in using third party script as a service under sandboxed environment (hiding Deno namespace) would be nice. We also want to be more web compatible)

**Comment by afinch7** [comment]
I think this is resolved after #2751, but I'm not sure.

**Comment by kevinkassimo** [comment]
Hmm I am still slightly in doubt, as this does not seem to actually address the nested worker case (they will always break the transparency and based on path of main module)

**Comment by bartlomieju** [comment]
Fixed in #3828
